plugins/by-name/faust/default.nix

```
@@ -0,0 +1,46 @@
+{ lib, pkgs, ... }:
+let
+  globalPrefix = "faust";
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "faust";
+  packPathName = "faust-nvim";
+  package = "faust-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # Configuration is done through globals vim.g.faust*
+  hasSettings = false;
+  extraOptions = {
+    faustPackage = lib.mkPackageOption pkgs "faust" {
+      nullable = true;
+    };
+
+    settings = lib.nixvim.plugins.vim.mkSettingsOption {
+      inherit globalPrefix;
+      name = "faust";
+      example = {
+        _examples_dir = "/usr/share/faust/examples";
+        "2appls_dir" = "/bin/";
+        lib_dir = "/usr/share/faust/";
+      };
+    };
+  };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    plugins.faust = {
+      luaConfig.content = ''
+        require('faust-nvim')
+        require('faust-nvim').load_snippets()
+      '';
+
+      settings = lib.mkIf (cfg.faustPackage != null) {
+        _examples_dir = "${cfg.faustPackage}/share/faust/examples";
+        "2appls_dir" = lib.getBin cfg.faustPackage;
+        lib_dir = "${cfg.faustPackage}/share/faust/";
+      };
+    };
+    globals = lib.nixvim.applyPrefixToAttrs globalPrefix (cfg.settings or { });
```

```suggestion
    globals = lib.nixvim.applyPrefixToAttrs globalPrefix cfg.settings;
```

The `settings` attr should always exist